internal/cmd/marketplace/apply/upload_image_test.go

```diff
@@ -0,0 +1,232 @@
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func TestApplyGetAndValidateImageLocalPath(t *testing.T) {
+	t.Run("should return local path if element contains image - YAML", func(t *testing.T) {
+		mockItemYAML := []byte(`---
+image:
+  localPath: ./someImage.png
+`)
+		mockItem := &marketplace.Item{}
+		err := yaml.Unmarshal(mockItemYAML, mockItem)
+		require.NoError(t, err)
+
+		found, err := getAndValidateImageLocalPath(mockItem, imageKey, imageURLKey)
+		require.NoError(t, err)
+		require.Equal(t, found, "./someImage.png")
+	})
+	t.Run("should throw error with an item that contains both image and imageURL", func(t *testing.T) {
+		mockItem := &marketplace.Item{
+			imageKey: map[string]interface{}{
+				localPathKey: "some/local/path/image.jpg",
+			},
+			imageURLKey: "http://some.url",
+		}
+
+		found, err := getAndValidateImageLocalPath(mockItem, imageKey, imageURLKey)
+		require.ErrorIs(t, err, errImageURLConflict)
+		require.Zero(t, found)
+	})
+
+	t.Run("should return local path if element contains image", func(t *testing.T) {
+		mockItemJSON := []byte(`{
+			"image": {
+				"localPath": "some/local/path/image.jpg"
+			}
+		}`)
+		mockItem := &marketplace.Item{}
+		err := yaml.Unmarshal(mockItemJSON, mockItem)
+		require.NoError(t, err)
+
+		found, err := getAndValidateImageLocalPath(mockItem, imageKey, imageURLKey)
+		require.NoError(t, err)
+		require.Equal(t, found, "some/local/path/image.jpg")
+	})
+
+	t.Run("should return error if image object is not valid", func(t *testing.T) {
+		mockItem := &marketplace.Item{
+			imageKey: map[string]interface{}{
+				"someWrongKey": "some/local/path/image.jpg",
+			},
+		}
+
+		found, err := getAndValidateImageLocalPath(mockItem, imageKey, imageURLKey)
+		require.ErrorIs(t, err, errImageObjectInvalid)
+		require.Zero(t, found)
+	})
+	t.Run("should not return anything if only imageUrl is found", func(t *testing.T) {
+		mockItem := &marketplace.Item{
+			imageURLKey: "http://some.url",
+		}
+
+		found, err := getAndValidateImageLocalPath(mockItem, imageKey, imageURLKey)
+		require.NoError(t, err)
+		require.Zero(t, found)
+	})
+}
+
+const mockImagePath = "./testdata/imageTest.png"
+
+type ErrReader struct {
+	err error
+}
+
+func (er *ErrReader) Read([]byte) (int, error) {
+	return 0, er.err
+}
+
+func TestApplyReadContentType(t *testing.T) {
+	t.Run("should read correct content type", func(t *testing.T) {
+		imageFile, err := os.Open(mockImagePath)
+		require.NoError(t, err)
+		defer imageFile.Close()
+		found, err := readContentType(imageFile)
+		require.NoError(t, err)
+		require.Equal(t, "image/png", found)
+	})
+
+	t.Run("should return error if read fails", func(t *testing.T) {
+		mockErr := errors.New("testing error")
+		found, err := readContentType(
+			&ErrReader{
+				err: mockErr,
+			},
+		)
+		require.ErrorIs(t, err, mockErr)
+		require.Zero(t, found)
+	})
+}
+
+func TestApplyUploadImage(t *testing.T) {
```

Comment:
Are we testing also the behavior in case of an error from the server?

Reply:
added tests